services/streammanager/kafka/client/producer.go

```diff
@@ -140,28 +138,45 @@ func headers(msg Message) (headers []kafka.Header) {
 }
 
 func isErrTemporary(err error) bool {
+	if errors.Is(err, context.DeadlineExceeded) {
+		return true
+	}
 	isTransientNetworkError := errors.Is(err, io.ErrUnexpectedEOF) ||
 		errors.Is(err, syscall.ECONNREFUSED) ||
 		errors.Is(err, syscall.ECONNRESET) ||
 		errors.Is(err, syscall.EPIPE)
 	if isTransientNetworkError {
 		return true
 	}
```

Comment:
```suggestion
	isTransientNetworkError := errors.Is(err, io.ErrUnexpectedEOF) ||
		errors.Is(err, syscall.ECONNREFUSED) ||
		errors.Is(err, syscall.ECONNRESET) ||
		errors.Is(err, syscall.EPIPE) || errors.Is(err, context.DeadlineExceeded)
	if isTransientNetworkError {
		return true
	}
```

Reply:
I did that way in the beginning but moved it in a separate `if` because a `context.DeadlineExceeded` could be due to things unrelated to network as well, so I don't want it to change the `isTransientNetworkError`.